server/src/index.ts

```diff
@@ -6,34 +6,46 @@ import cors from "cors";
 const app = express();
 const port = 4000;
 const server = app.listen(port, () => {
-  return console.log(`server is listening on ${port}`);
+	return console.log(`server is listening on ${port}`);
 });
 
 const io = new Server(server, {
-  cors: {
-    origin: ["http://localhost:4500"],
-  },
+	cors: {
+		origin: ["http://localhost:4500"],
+	},
 });
 
 app.get("/", (req, res) => {
-  res.send({ body: "Hello world, 3" });
+	res.send({ body: "Hello world, 3" });
 });
 
-const users = [{}];
+const users: object[] = [];
 io.on("connection", (socket) => {
-  console.log("new client: " + socket.id);
-  socket.on("NEW_USER", (name) => {
-    users.push({ id: socket.id, name });
-    socket.broadcast.emit("NEW_USER", name);
-  });
-  socket.on("OFFER", (dataString) => {
-    socket.broadcast.emit("OFFER", dataString);
-  });
-  socket.on("ANSWER", (dataString) => {
-    const target = JSON.parse(dataString).target;
-    socket.broadcast.to(target).emit("ANSWER", dataString);
-  });
-  socket.on("disconnect", () => {
-    console.log("client disconnect");
-  });
+	console.log("new client: " + socket.id);
+	socket.on("REQUEST_USERS", () => {
+		console.log("users requested by: " + socket.id);
+		socket.emit("REQUEST_USERS", users);
+	});
+	socket.on("JOIN", (name) => {
+		const user = { name, id: socket.id };
+		users.push(user);
+		console.log("new user pushed: " + socket.id);
+		io.emit("JOIN", user);
+	});
+	socket.on("OFFER", (dataString) => {
+		const targetId = JSON.parse(dataString).receiverId;
+		socket.broadcast.to(targetId).emit("OFFER", dataString);
+	});
+	socket.on("ANSWER", (dataString) => {
+		const targetId = JSON.parse(dataString).receiverId;
+		socket.broadcast.to(targetId).emit("ANSWER", dataString);
+	});
+	socket.on("disconnect", () => {
```

Comment:
Make this uppercase instead of lower case to match the other ones?

Reply:
"connect" and "disconnect" are built in events so I don't think I can change them. Will consolidate the event names into a separate constants file and make them lowercase in the refactor branch.